source/adios2/core/ADIOS.cpp

```diff
@@ -79,6 +79,18 @@ IO &ADIOS::DeclareIO(const std::string ioName)
     return ioPair.first->second;
 }
 
+IO &ADIOS::GetIO(const std::string name)
+{
+    auto itIO = m_IOs.find(name);
+    if (itIO == m_IOs.end())
```

Comment:
This is the kind of discussion that end with testing code. Yeah, dereferencing is undefined behavior that's won't show until you try accessing the dereferenced iterator within C++. That's when the exception will be thrown, typically as bad_alloc.  Anyways, the point here is that we don't guarantee anything if m_DebugMode == false. That's the user being confident on its own flow.